### ISSUE TITLE
Functions in SQL

### ISSUE THREAD

**Issue by pgulutzan** [tarantool/tarantool#2347]
Possible problems with functions in SQL statements.

1. ABS(-9223372036854775807) returns -1.
```
box.sql.execute("VALUES (ABS(-9223372036854775807));")
`````
Result: [-1]

2. ABS(-9223372036854775808) returns nothingL.
```
box.sql.execute("VALUES (ABS(-9223372036854775808));")
```
Result: []

3. String functions GLOB, LIKE(), and LENGTH() stop when
'\0' exists in the input, if data type = text.
This is not supposed to be an end-of-string signal in SQL.
```
box.sql.execute("VALUES (GLOB(CHAR(65,00,65),'A'), LENGTH(CHAR(65,00,65)));")
````
Result: [1, 1]

4. LTRIM(X'004100',X'00') does not trim X'00'.
Finding X'00' in a BLOB seems to be a general problem.
I'm not really asking for a conversion,
I'm asking for trimming to work with BLOB.
(I'm still uncertain how the BLOB data
type should work, but if it exists,
LTRIM should work the way that TRIM(LEADING
works in standard SQL -- or so I thought.)
```
box.sql.execute("VALUES (LTRIM(X'004100',X'00'));")
````
Result: ["\0A\0"]

5. RANDOMBLOB(0) returns a 1-byte string. SQLite does too,
but that doesn't mean it makes sense. The argument for
RANDOMBLOB is supposed to be the number of bytes.
```
box.sql.execute("VALUES (LENGTH(RANDOMBLOB(0)));")
````
Result: [1]

6. ROUND(1e-31,32) returns 0.
```
box.sql.execute("VALUES (ROUND(1e-31,32));")
````
Result: [0]

7. TYPEOF(randomblob(5) || zeroblob(5)) is 'text'.
This is SQLite behaviour -- || always causes text
or null results -- but once again it seems to be a
rule that depends on the operator, not the operands.
```
box.sql.execute("VALUES (TYPEOF(randomblob(5) || zeroblob(5)));")
```
Result = 'text'

**Comment by kyukhin** [comment]
@pgulutzan, what we have today:
- 1 and 2 seems to be resolved
- IMHO, 3 worse separate issue (for 2.1.1)
- Same for 4 trim() routine always convert input argument to text, so it needs deep refactoring
- 5 and 6 are fixed on branch and hopefully will hit 2.1.0 branch
- 7 goes w/ 3 and 4: separate ticket (for 2.1.1) and deep refactoring of corresponding opcode.

So, if no objections, I'll submit 3 new issues (for 3. 4. and 7.) and will close this, once branch
is reviewed and merged into 2.1.0 branch.

**Comment by pgulutzan** [comment]
(1) is fixed. While checking this, I noticed that
"select (9223372036854775809);" returns 9223372036854775808,
so I think there is something wrong still with the extreme
limit of integers. Perhaps you already knew that.

(2) returns "integer overflow". Is that the right result?
I think you had a choice here: you could have promoted to
a float instead of returning an error. Perhaps you already
decided that is not what users want.

(3) and (4) and (5) and (6) and (7) are not fixed in 2.0, but
you are making new issues or fixing them in 2.1, which should
make everyone happy. After all, these are minor issues.

**Comment by kyukhin** [comment]
@pgulutzan, concerning (1) saturation of big integers and (2) error reporting on big integers in abs().

I believe that promoting int to float is really dangerous. Moreover, silently saturating to MAX/MIN_INT
is even more dangerous. So, I am going to report error when constant cannot fix in MAX/MIN_INT range.
For literals and for built-in functions like abs().
